src/bridge/IrcBridge.ts

```diff
@@ -803,6 +808,34 @@ export class IrcBridge {
         request.outcomeFrom(this._onEvent(request));
     }
 
+    private onEphemeralEvent(request: Request<EphemeralEvent>) {
+        // If we see one of these events over federation, bump the
+        // last active time for those users.
+        const event = request.getData();
+        let userIds: string[]|undefined = undefined;
+        if (!this.activityTracker) {
+            return;
+        }
+        if (event.type === "m.presence" && event.content.presence === "online") {
+            userIds = [event.sender];
+        }
+        else if (event.type === "m.receipt") {
+            userIds = Object.keys(Object.values(event.content).map((v) => v["m.read"]));
```

Comment:
Is this taking ALL the `m.read`s? Surely we still only want the sender in this point? Or you refresh the whole list whenever there's a new `m.receipt`?

Reply:
Synapse sends new receipts in the format https://matrix.org/docs/spec/client_server/r0.6.0#m-receipt, which makes it a bit tricky to determine what's changed. We *could* do some quick filtering to exclude *stale* receipts though.

Reply:
I've tried something else.